### PR TITLE
Support vs2015, Fix race condition when building python

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

File: Libs/MRML/Core/Testing/vtkMRMLSceneTest2.cxx [hunk 1]
```diff
@@ -23,6 +23,9 @@
 
 // STD includes
 #include <algorithm>
+#if _MSC_VER >= 1900
```
Comment (msmolens) [inline Libs/MRML/Core/Testing/vtkMRMLSceneTest2.cxx]: The  `_MSC_VER` check isn't necessary.

File: Modules/Scripted/DataProbe/Logic/vtkPVScalarBarActor.cxx [hunk 2]
```diff
@@ -211,7 +211,7 @@ int vtkPVScalarBarActor::CreateLabel(
   // One Windows, formats with exponents have three digits by default
   // whereas on other systems, exponents have two digits. Set to two
   // digits on Windows for consistent behavior.
-#ifdef _WIN32
+#if _WIN32 && _MSC_VER < 1900
```
Comment (msmolens) [inline Modules/Scripted/DataProbe/Logic/vtkPVScalarBarActor.cxx]: Let's fix this more like was done in Paraview: https://github.com/Kitware/ParaView/commit/d4f2b29b4d63ae3dfd8553ee98a4d5eb181e8244

**Comment by jcfr** [comment]
@msmolens  Thanks for the review

File: Modules/Scripted/DataProbe/Logic/vtkPVScalarBarActor.cxx [hunk 3]
```diff
@@ -208,10 +208,10 @@ int vtkPVScalarBarActor::CreateLabel(
     textActor->ComputeScaledFont(this->P->Viewport);
     }
 
-  // One Windows, formats with exponents have three digits by default
+  // On Visual Studio prior 2015, formats with exponents have three digits by default
   // whereas on other systems, exponents have two digits. Set to two
   // digits on Windows for consistent behavior.
-#ifdef _WIN32
+#if defined(_MSV_VER) && (_MSC_VER < 1900)
```
Comment (msmolens) [inline Modules/Scripted/DataProbe/Logic/vtkPVScalarBarActor.cxx]: typo: `_MSC_VER  `is misspelled

Comment (jcfr) [inline Modules/Scripted/DataProbe/Logic/vtkPVScalarBarActor.cxx]: Ooops, I blindly copied the paraview code. This is now fixed in the trunk

**Comment by jcfr** [comment]
Topic integrated in [r26188](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=26188), [r26189](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=26189), [r26190](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=26190) and [r26191](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=26191)

**Pull request closed without merging**